### ISSUE TITLE
PyPI build failure on macOS Sonoma 14.2.1

### ISSUE THREAD

**Issue by externl** [zeroc-ice/ice#1734]
A user reported the following:

Hi,

I tried to install zeroc-ice by running python3 -m pip install zeroc-ice on MacOS Sonoma 14.2.1, and it stuck on "Building wheel for zero-ice (pyproject.toml) ... for a very long time, and then got the following error message.

src/ice/mcpp/system.c:1009:20: error: call to undeclared function 'readlink'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
if ((len = readlink( slbuf1, slbuf2, PATHMAX)) > 0) {
^
src/ice/mcpp/system.c:1184:20: error: call to undeclared function 'readlink'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
if ((len = readlink( slbuf1, slbuf2, PATHMAX)) > 0) {
^
2 errors generated.
error: command '/usr/bin/clang' failed with exit code 1
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for zeroc-ice
Running setup.py clean for zeroc-ice
Failed to build zeroc-ice
ERROR: Could not build wheels for zeroc-ice, which is required to install pyproject.toml-based projects

**Comment by externl** [comment]
Temporary workaround
```
CFLAGS='-Wno-implicit-function-declaration' python3 -m pip install zeroc-ice
```

**Comment by externl** [comment]
We should also check the npm slice2js build.

**Comment by pepone** [comment]
This was fixed in 3.7.10.1 PyPI package https://pypi.org/project/zeroc-ice/3.7.10.1/
